packages/@lwc/engine-dom/src/apis/build-custom-element-constructor.ts

```diff
@@ -120,6 +124,23 @@ export function buildCustomElementConstructor(Ctor: ComponentConstructor): HTMLE
             attributeChangedCallback.call(this, name, oldValue, newValue);
         }
 
+        formAssociatedCallback() {
+            runFormAssociatedCallback(this);
+        }
+
+        formDisabledCallback() {
+            runFormDisabledCallback(this);
+        }
+
+        formResetCallback() {
+            runFormResetCallback(this);
+        }
+
+        formStateRestoreCallback() {
+            runFormStateRestoreCallback(this);
+        }
```

Comment:
The face lifecycle methods should only be available with native lifecycle. 

Since `CustomElementConstructor` will be directly invoked by the browser, it is technically always using the native lifecycle.

Reply:
Correct, yes, it looks like `CustomElementConstructor` is inherently always in native-lifecycle mode. I hadn't considered this before!